subprojects/docs/src/docs/userguide/authoring-builds/more_about_tasks.adoc

```diff
@@ -743,6 +743,20 @@ There are other task types where this kind of nested inputs are available:
 
 In the same way, this kind of modelling is available to custom tasks.
 
+[[sec:task_input_validation]]
+===== Runtime validation
+
+When executing the build Gradle checks if task types are declared with the proper annotations.
```

Comment:
Can we say a little bit how/who needs to fix the problems, given that this is what the release notes suggest?

Reply:
Do you have any concrete suggestion? Adding "the problems need to be fixed by the author of the task or plugin in question" doesn't seem to add much.

Reply:
I would add the same text as in the upgrading notes:
```
If you own the code of the tasks in question, you can fix them by following the instructions.
Otherwise you'll need to report the problems to the maintainer of the relevant task or plugin.
```

I wonder if we should also link to some documentation about deprecation warnings in the user guide.